### PR TITLE
Update epoch_metric and recall in test for generating data with different rank

### PR DESCRIPTION
**Pull request by puhuk**
Description: Update `epoch_metric` and `recall`

Check list:

- [x] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: tests/ignite/metrics/test_recall.py [hunk 1]
```diff
@@ -540,7 +544,8 @@ def update(engine, i):
     metric_devices = ["cpu"]
     if device.type != "xla":
         metric_devices.append(idist.device())
-    for _ in range(2):
+    for i in range(2):
+        torch.manual_seed(12 + rank + i)
```
Comment (sadra-barikbin) [inline tests/ignite/metrics/test_recall.py:548]: So you could remove line 497 and bring line 496 near here.

File: tests/ignite/metrics/test_recall.py [hunk 2]
```diff
@@ -475,7 +476,8 @@ def update(engine, i):
     metric_devices = [torch.device("cpu")]
     if device.type != "xla":
         metric_devices.append(idist.device())
-    for _ in range(2):
+    for i in range(2):
```
Comment (sadra-barikbin) [inline tests/ignite/metrics/test_recall.py:479]: So you could bring line 433 near here.

File: tests/ignite/metrics/test_recall.py (hunk 1 above)
Comment (vfdev-5) [reply tests/ignite/metrics/test_recall.py:548]: Good catch @sadra-barikbin! 

@puhuk please ensure that you do not leave `torch.manual_seed(12)` in other metric tests

File: tests/ignite/metrics/test_recall.py [hunk 3]
```diff
@@ -545,6 +543,7 @@ def update(engine, i):
     if device.type != "xla":
         metric_devices.append(idist.device())
     for i in range(2):
+        rank = idist.get_rank()
```
Comment (sadra-barikbin) [inline tests/ignite/metrics/test_recall.py:546]: Could you please put this outside the `for` loop?

File: tests/ignite/metrics/test_recall.py [hunk 4]
```diff
@@ -477,6 +475,7 @@ def update(engine, i):
...
```
Comment (sadra-barikbin) [inline tests/ignite/metrics/test_recall.py:478]: Could you please put this outside the for loop?

**Review by sadra-barikbin — APPROVED** [review]

**Pull request merged**